File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:

        map1={}


        for i in s:
            map1[i] = map1.get(i,0)+1
        
        for j in t:
            if j in map1:
                map1[j]-=1
                if map1[j]==0:
                    del map1[j]
            else:
                return False
        return len(map1) ==0

        

File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {

        HashMap<Character,Integer> map = new HashMap();


        for(char i : s.toCharArray()){

            map.put(i,map.getOrDefault(i,0)+1);
        }

        for(char ch:t.toCharArray()){

            if (map.containsKey(ch)){

               
                

                map.put(ch,map.get(ch)-1);

                 if(map.get(ch)==0){
                    map.remove(ch);
                }
                

            }
            else{

                return false;
            }


        }

        System.out.print(map.values());

        return map.size()==0;
        
    }
}